ddl-introduction.md

```diff
@@ -0,0 +1,183 @@
+
+### Types of DDL statements
+
+Based on whether a DDL statement blocks the user application during execution, DDL statements can be divided into the following types:
+
+- **Offline DDL statements**: When the database receives a DDL statement from the user, it first locks the database object to be modified, executes the metadata change, and blocks the user application from modifying data during the DDL execution.
+
+- **Online DDL statements**: When a DDL statement is executed in the database, a specific method is used to ensure that the statement does not block the user application. This allows the user to submit modifications during the DDL execution. The method also ensures the correctness and consistency of the corresponding database object during the execution process.
+
+Based on whether to operate the data included in the target DDL object, DDL statements can be divided into the following types:
+
+- **Logical DDL statements**: This type of DDL statements usually only modify the metadata of the database object, without processing the data stored in the object. For example, changing the table name or changing the column name.
+
+    In TiDB, logical DDL statements are also referred to as "general DDL". These statements typically have a short execution time, often taking only a few tens of milliseconds or seconds to complete. As a result, they do not consume much system resource and do not affect the load on the application.
+
+- **Physical DDL statements**: This type of DDL statements not only modifies the metadata of the object to be changed, but also modifies the user data stored in the object. For example, when TiDB creates an index for a table, it not only changes the definition of the table, but also performs a full table scan to build the newly added index.
+
+    In TiDB, physical DDL statements are also referred to as "reorg DDL", which stands for reorganization. Currently, physical DDL statements only include `ADD INDEX` and lossy column type changes (such as changing from an `INT` type to a `CHAR` type). These statements take a long time to execute, and the execution time is influenced by the amount of data in the table, the machine configuration, and the application load.
+
+    Executing physical DDL statements can have an impact on the load of the application for two reasons. On the one hand, it requires CPU and I/O resources from TiKV to read data and write new data. On the other hand, the TiDB node where the DDL Owner is located needs to perform the corresponding computations, which consumes more CPU resources. Because TiDB does not currently support distributed execution of DDL statements, other TiDB nodes do not consume additional system resources during this process.
```

Comment:
```suggestion
    Executing physical DDL statements can have an impact on the wokrload of the application for two reasons. On the one hand, it consumes CPU and I/O resources from TiKV to read data and write new data. On the other hand, the TiDB node where the DDL Owner is located needs to perform the corresponding computations, which consumes more CPU resources. Because TiDB does not support distributed execution of DDL statements, other TiDB nodes do not consume additional system resources during this process.
```